### ISSUE TITLE
[Idea] Create snapshots as quickly as possible and in parallel

### ISSUE THREAD

**Issue by schildbach** [digint/btrbk#436]
Currently the entire btrbk.conf is executed sequentially, with each target iterating through the several phases. This means creating snapshots can be delayed by quite a bit (hours in my case), with their names lying about their creation time.

My idea is if you use any action that involves creating snapshots, **the phase of creating snapshots is executed before anything else, ideally in parallel so each of the snapshots is created as quickly as possible**. Only if that phase finishes, sequential execution would continue for the other phases (although those could be parallelized too – out of scope for this issue).

Optional: Maybe the creation of snapshots could be moved outside of the btrbk global locking, so a long-running btrbk would at least not prevent snapshot creation of the next run.

**Comment by bdaase** [comment]
Sending them to different destinations in parallel would also be good. I just suggested it here https://github.com/digint/btrbk/issues/214#issuecomment-997161015.

**Comment by digint** [comment]
I'm not sure what you are referring to, as btrbk always runs snapshots at first:
https://digint.ch/btrbk/doc/btrbk.1.html#_actions

Feel free to reopen the issue if I'm missing something...

**Comment by schildbach** [comment]
If I interpret the console output correctly, then yes it runs snapshots first for each volume. But if you've got two volumes it will snapshot the second volume after all the phases of the first volume have been finished. This can be hours after the timestamp that btrbk uses in the name of the snapshot.
